### PR TITLE
`dependencies` Update `digitaltwins` api version to 2020-12-01

### PR DESCRIPTION
**Pull request by myc2h6o**
- dependency update for #15970
- `authenticationType` is newly added and required, set to the default `keyBased`
- `client.Update` in `resourceDigitalTwinsInstanceUpdate` is changed in the new API to return `future`, updated accordingly.

### CODE REVIEW

File: internal/services/digitaltwins/digital_twins_instance_resource.go [hunk 1]
```diff
@@ -149,10 +149,15 @@ func resourceDigitalTwinsInstanceUpdate(d *pluginsdk.ResourceData, meta interfac
...
 	}
 
-	if _, err := client.Update(ctx, id.ResourceGroup, id.Name, props); err != nil {
+	future, err := client.Update(ctx, id.ResourceGroup, id.Name, props)
+	if err != nil {
 		return fmt.Errorf("updating Digital Twins Instance %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for update of the Digital Twins Instance %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```
Comment (catriona-m) [inline internal/services/digitaltwins/digital_twins_instance_resource.go:158]: These log messages could be simplified a bit
```suggestion
		return fmt.Errorf("updating %s: %+v", *id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", *id, err)
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @myc2h6o. Left one minor comment inline, otherwise this is looking good!

**Comment by myc2h6o** [comment]
Hi @catriona-m thanks for reviewing the change! I've resolved the comment, please take a look

**Review by catriona-m — APPROVED** [review]
Thanks @myc2h6o LGTM!

**Pull request merged**